### ISSUE TITLE
Gemma2 CCE doesn't actually apply softcapping

### ISSUE THREAD

**Issue by winglian** [apple/ml-cross-entropy#48]
tracing through the Gemma2 modeling patch, it never enables softcapping for the cross entropy loss.  It's not exposable either through PatchOptions.

**Comment by winglian** [comment]
I think https://github.com/apple/ml-cross-entropy/blob/main/cut_cross_entropy/transformers/gemma2.py#L108 needs to be modified to 

```
loss = apply_lce(hidden_states, self.lm_head.weight, labels, _PATCH_OPTS, softcap=True, **loss_kwargs)
```
